notify/drivers/salesforce/salesforce.py

```diff
@@ -0,0 +1,390 @@
+            return True
+        else:
+            LOG.debug("Unexpected error: Case was not created (code != 201) "
+                      "and Case does not exist (code != 400)")
+            return False
+
+    except requests.exceptions.ConnectionError as e:
+        LOG.debug(e)
+        LOG.debug('Unexpected error: Case was not created: Connection error.')
+        return False
+
+
+class OAuth2(object):
+    def __init__(self,
+                 client_id,
+                 client_secret,
+                 username,
+                 password,
+                 auth_url=None,
+                 organizationId=None):
+
+        if not auth_url:
+            auth_url = 'https://login.salesforce.com'
+
+        self.auth_url = auth_url
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.username = username
+        self.password = password
+        self.organizationId = organizationId
+
+    def getUniqueElementValueFromXmlString(self, xmlString, elementName):
+        """Extracts an element value from an XML string.
+
+        For example, invoking
+        getUniqueElementValueFromXmlString('
+           <?xml version="1.0" encoding="UTF-8"?><foo>bar</foo>', 'foo'
+        )
+        should return the value 'bar'.
+        """
+        xmlStringAsDom = xml.dom.minidom.parseString(xmlString)
+        elementsByName = xmlStringAsDom.getElementsByTagName(elementName)
+        elementValue = None
+        if len(elementsByName) > 0:
+            toXML = elementsByName[0].toxml()
+            r1 = '<' + elementName + '>'
+            r2 = '</' + elementName + '>'
+            elementValue = toXML.replace(r1, '').replace(r2, '')
+        return elementValue
+
+    def authenticate_soap(self):
+
+        soap_url = '{}/services/Soap/u/36.0'.format(self.auth_url)
+
+        login_soap_request_body = """<?xml version="1.0" encoding="utf-8" ?>
+        <soapenv:Envelope
+                xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/"
+                xmlns:urn="urn:partner.soap.sforce.com">
+            <soapenv:Header>
+                <urn:CallOptions>
+                    <urn:client>RestForce</urn:client>
+                    <urn:defaultNamespace>sf</urn:defaultNamespace>
+                </urn:CallOptions>
+                <urn:LoginScopeHeader>
+                    <urn:organizationId>{organizationId}</urn:organizationId>
+                </urn:LoginScopeHeader>
+            </soapenv:Header>
+            <soapenv:Body>
+                <urn:login>
+                    <urn:username>{username}</urn:username>
+                    <urn:password>{password}</urn:password>
+                </urn:login>
+            </soapenv:Body>
+        </soapenv:Envelope>""".format(username=self.username,
+                                      password=self.password,
+                                      organizationId=self.organizationId)
```

Comment:
Is this possible that XML entities can appear in self.username, self.password and self.organizationId?

Example:

   self.password is "pass<>word"
   self.organizationId is "Foo & Bar" or "Foo's Bar"

If yes, these values must be sanitized, otherwise XML will be invalid